src/server/models/listItem/listItem.ts

```diff
@@ -105,33 +105,32 @@ export async function findListItemById(id: string | number) {
   }
 }
 
-export async function findListItems(options: {
-  listIds?: number[];
-  listItemIds?: number[];
-  statuses?: Status[];
-  isAnnualReview?: boolean;
-}) {
+export async function findListItems(options: { listIds: number[]; statuses?: Status[]; isAnnualReview?: boolean }) {
   try {
-    const { listIds, listItemIds, statuses, isAnnualReview } = options;
-    if ((!listIds || !listIds.length) && (!listItemIds || !listItemIds?.length)) {
+    const { listIds, statuses, isAnnualReview } = options;
+    if (!listIds?.length) {
       const message = "List ids or list item ids must be specified to find list items";
       logger.error(message);
-      return { error: Error(message) };
+      return { error: message };
     }
+
+    const tableData = await Promise.all(listIds.map(async (listId) => ({ list: await findListById(listId), listId })));
+
     const result = await prisma.listItem.findMany({
       where: {
-        ...(listIds != null && { listId: { in: listIds } }),
-        ...(listItemIds != null && { id: { in: listItemIds } }),
-        ...(statuses != null && { status: { in: statuses } }),
-        ...(isAnnualReview != null && { isAnnualReview }),
-        history: {
-          some: {
-            type: "PUBLISHED",
-            time: {
-              lte: subMonths(Date.now(), 1),
+        AND: tableData.map(({ list, listId }: any) => ({
+          listId,
+          ...(statuses != null && { status: { in: statuses } }),
+          ...(isAnnualReview != null && { isAnnualReview }),
+          history: {
+            some: {
+              type: "PUBLISHED",
+              time: {
+                lte: list?.jsonData?.currentAnnualReview?.keyDates.POST_ONE_MONTH,
```

Comment:
this condition fails for the batch process as the currentAnnualReview key hasn't been populated yet.  You could probably adjust to something like:

```suggestion
      where: {
        OR: tableData.map(({ list, listId }: any) => {
          const oneMonthBeforeAnnualReviewStart = subMonths(list?.nextAnnualReviewStartDate, 1);
          return ({
            listId,
            ...(statuses != null && { status: { in: statuses } }),
            ...(isAnnualReview != null && { isAnnualReview }),
            history: {
              some: {
                type: "PUBLISHED",
                time: {
                  lte: oneMonthBeforeAnnualReviewStart,
                },
              },
            },
          });
        }),
      },```

Reply:
Ah to be honest, I was meaning to revert this code so it shouldn't be there 🤦

Reply:
But actually... I think your suggestion makes more sense than reverting 🤔

Reply:
Ideally we aren't running a map in here. my suggestion here https://cautionyourblast.slack.com/archives/C03RT4ZR8E7/p1678817907977909

but essentially I don't think an additional findListItems is necessary if we get `findListByAnnualReviewDate` to return the right list items.